spec/controlling-tasks.md

```diff
@@ -0,0 +1,117 @@
+accept an {{AbortSignal}}. Additionally, {{Scheduler/postTask()}} accepts an
+{{AbortSignal}}, which can be useful if dynamic prioritization is not needed.
+
+The `TaskPriorityChangeEvent` Interface {#sec-task-priority-change-event}
+---------------------
+
+<pre class='idl'>
+  [Exposed=(Window, Worker)]
+  interface TaskPriorityChangeEvent : Event {
+    constructor(DOMString type, TaskPriorityChangeEventInit priorityChangeEventInitDict);
+
+    readonly attribute TaskPriority previousPriority;
+  };
+
+  dictionary TaskPriorityChangeEventInit : EventInit {
+    required TaskPriority previousPriority;
+  };
+</pre>
+
+A {{TaskPriorityChangeEvent}} object has a <dfn for=TaskPriorityChangeEvent>previousPriority</dfn>
+attribute. The {{TaskPriorityChangeEvent/previousPriority}} getter steps are to
+return [=this=]'s [=TaskPriorityChangeEvent/previousPriority=].
+
+The `TaskController` Interface {#sec-task-controller}
+---------------------
+
+<pre class='idl'>
+  [Exposed=(Window,Worker)]
+  interface TaskController : AbortController {
+    constructor(optional TaskPriority priority = "user-visible");
+
+    undefined setPriority(TaskPriority priority);
+  };
+</pre>
+
+Note: {{TaskController}}'s {{AbortController/signal}} getter, which is
+inherited from {{AbortController}}, returns a {{TaskSignal}} object.
+
+<div algorithm>
+  The <dfn constructor for="TaskController" lt="TaskController()"><code>new TaskController(|priority|)</code></dfn>
+  constructor steps are:
+
+  1. Let |signal| be a new {{TaskSignal}} object.
+  1. Set |signal|'s <a for=TaskSignal>priority</a> to |priority|.
+  1. Set [=this's=] <a for=AbortController>signal</a> to |signal|.
+</div>
+
+The <dfn method for=TaskController><code>setPriority(|priority|)</code></dfn>
+method steps are to <a for=TaskSignal>signal priority change</a> on [=this=]'s
+<a for=AbortController>signal</a> given |priority|.
+
+The `TaskSignal` Interface {#sec-task-signal}
+---------------------
+
+<pre class='idl'>
+  [Exposed=(Window, Worker)]
+  interface TaskSignal : AbortSignal {
+    readonly attribute TaskPriority priority;
+
+    attribute EventHandler onprioritychange;
+  };
+</pre>
+
+A {{TaskSignal}} object has an associated {{TaskPriority}}
+<dfn for=TaskSignal>priority</dfn>.
+
+A {{TaskSignal}} object has an associated <dfn attribute for=TaskSignal>priority changing</dfn>
+flag. It is unset unless otherwise specified.
+
+A {{TaskSignal}} object has associated <dfn attribute for=TaskSignal>priority change algorithms</dfn>,
+which is a [=set=] of algorithms which are to be executed when its
+{{TaskSignal/priority changing}} flag is set. Unless specified otherwise, its value is
+the empty set.
+
+The <dfn attribute for="TaskSignal">priority</dfn> getter steps are to return [=this=]'s <a for=TaskSignal>priority</a>.
+
+The <dfn attribute for=TaskSignal><code>onprioritychange</code></dfn> attribute
+is an [=event handler IDL attribute=] for the {{TaskSignal/onprioritychange}}
```

Comment:
Linking back to your own definition is probably not right. I'm not clear what the best pattern is here, but maybe just non-linked code font (backticks) for this `onprioritychange`

Reply:
Eh... I mean, nobody's going to implement the wrong thing based on the Page Lifecycle spec, so in some sense it's good enough. So either way works IMO.